orchagent/vnetorch.cpp

```diff
@@ -2804,13 +2858,33 @@ bool VNetRouteOrch::handleTunnel(const Request& request)
         SWSS_LOG_INFO("Handling Priority Tunnel with prefix %s", ip_pfx.to_string().c_str());
     }
 
-    NextHopGroupKey nhg_primary("", true);
-    NextHopGroupKey nhg_secondary("", true);
+    if (primary_list.size() != ip_list.size())
+    {
+        for (auto& ip : primary_list)
+        {
+            if (std::find(ip_list.begin(), ip_list.end(), ip) == ip_list.end())
+            {
+                secondary_list.push_back(ip);
+            }
+        }
+    }
+
+    if (check_directly_connected &&  (isPartiallyLocal(primary_list) || isPartiallyLocal(secondary_list) ))
+    {
+        SWSS_LOG_ERROR("Endpoints in Primary/backup should either all be local endpoints or no local endpoint at all.");
+        return false;
+    }
+
+    NextHopGroupKey nhg_primary("", primary_list.empty() || !isLocalEndpoint(vnet_name, primary_list[0]));
```

Comment:
Can we retain the old code and put this as else of `if (check_directly_connected)`